### ISSUE TITLE
Context error when emitting event from "set"

### ISSUE THREAD

**Issue by JeoCo** [IndigoSoft/ngxd#21]
Hello, I've [forked a stackblitz](https://stackblitz.com/edit/angular-ngxd-example-with-output-binding-jkxzrn?file=src/app/hello.component.ts) from one of the other (solved) issues here. 

I was trying to create a very basic example where the "name" property in the "hello" component is emitting an event every time a "set" occurs (so one could keep track of its value in its parent component).

The following is all I changed: ( explanations in comments)
```
export class HelloComponent  {
  @Output() myClick: EventEmitter<string> = new EventEmitter<string>();
  @Output() setEvent: EventEmitter<string> = new EventEmitter<string>(); // added a second emitter
  component = DynamicComponent;
  
  private _name: string; // made name private
  @Input() set name(value : string){
    console.log('set name', value, this); // log so you can see the error and context
    this._name = value; 
    this.setEvent.emit(value); // emitting the event so the parent component (in this case app component) can listen to it and keep track of the value
  }
  get name(){ return this._name;}
  
}  
```
I then encounter a very strange phenomenon where "set name()" is run multiple times (without any actual need) with the last of them having an empty context ( this = {} ) thus throwing the following errors: 

> ERROR TypeError: Cannot read property 'emit' of undefined

> ERROR CONTEXT

I should note that I'm specifically interested in the "name" property (that comes via @Input) - the goal is to keep its value synced with the parent in case it changes.

Also I should mention that if you give DynamicComponent a selector and place it directly in your template the issue doesn't occur. I've tested thoroughly seems to be an issue with ngxd.  

Could you maybe shed some light on this behaviour? What am I doing wrong?
Thanks :)

**Comment by thekiba** [comment]
Hello! I've seen the bug on a stackblitz link, but I can't repeat it in the tests.

Can you make Pull Request with the failed test? And then I'll fix it asap.

**Comment by davidhunt135** [comment]
I have this same problem but I'm not sure on reproducing. It happens whatever I try

**Comment by alQlagin** [comment]
Hi, @davidhunt135 
I'he reproduced it and added pr #24 with test

**Comment by thekiba** [comment]
I published new 9.0.4 version. It should to fix that.
@JeoCo @davidhunt135 @alQlagin Thanks 💖
